### DISCUSSION TITLE
Excessive scheduler restarts after migrating from version 2.5.1 to 2.6.0.

### DISCUSSION THREAD

**Discussion by thaysa** [apache/airflow#31070] [Q&A] [answered]
The issue started after migrating to Airflow 2.6.0. I am seeing the following error: '**Liveness probe failed: Traceback (most recent call last): File "<string>", line 4, in <module> ModuleNotFoundError: No module named 'airflow.jobs.scheduler_job**'.'

The Docker image being used for Airflow is based on the tag 2.6.0-python3.9.

airflow.jobs.scheduler_job has been removed:

https://github.com/apache/airflow/blob/2.5.1/airflow/jobs/scheduler_job.py

**Comment by potiuk** [comment, marked as answer]
Your livenses probe uses some custom (or simply old) way of checking for liveness check. 

You can see that in the Helm chart of ours https://github.com/apache/airflow/blob/9720705d9281c8f13d96f4165fab77264306d517/chart/templates/_helpers.yaml#L649 

The (very old from pre 2.1.0) way was  this:
```
  - sh
  - -c
  - |
    CONNECTION_CHECK_MAX_COUNT=0 exec /entrypoint python -Wignore -c "
    import os
    os.environ['AIRFLOW__CORE__LOGGING_LEVEL'] = 'ERROR'
    os.environ['AIRFLOW__LOGGING__LOGGING_LEVEL'] = 'ERROR'
    from airflow.jobs.scheduler_job import SchedulerJob
    from airflow.utils.db import create_session
    from airflow.utils.net import get_hostname
    import sys
    with create_session() as session:
        job = session.query(SchedulerJob).filter_by(hostname=get_hostname()).order_by(
            SchedulerJob.latest_heartbeat.desc()).limit(1).first()
    sys.exit(0 if job.is_alive() else 1)"
```

Then in 2.1.0 it's changed to:

```
  - sh
  - -c
  - |
    CONNECTION_CHECK_MAX_COUNT=0 AIRFLOW__LOGGING__LOGGING_LEVEL=ERROR exec /entrypoint \
    airflow jobs check --job-type SchedulerJob --hostname $(hostname)
```

As of 2.5.0 it is:

```
  - sh
  - -c
  - |
    CONNECTION_CHECK_MAX_COUNT=0 AIRFLOW__LOGGING__LOGGING_LEVEL=ERROR exec /entrypoint \
    airflow jobs check --job-type SchedulerJob --local
```

You can see details here https://github.com/apache/airflow/blob/9720705d9281c8f13d96f4165fab77264306d517/chart/templates/_helpers.yaml#L649

So you should fix it in your Helm Char to use the latest version.

**Comment by thaysa** [comment]
Thank you very much, @potiuk ,  for your prompt response and for all the details. The scheduler is no longer restarting after fixing the helmchart.

**Reply by potiuk** [reply]
Glad I could help :)
